src/com/ms/silverking/thread/ThreadUtil.java

```diff
@@ -1,158 +1,155 @@
-        }
-    }
-    
-    public static void await(Condition cv, int millis) {
-        await(cv, millis, TimeUnit.MILLISECONDS);
-    }
-    
-    public static void randomAwait(Condition cv, int minMillis, int maxMillis) {
-        int    millis;
-        
-        if (maxMillis < minMillis) {
-            throw new RuntimeException("maxMillis < minMillis");
-        }
-        millis = minMillis + random.nextInt(Math.max(maxMillis - minMillis, 1));
-        await(cv, millis);
-    }
-    
-    public static void randomAwait(Condition cv, int maxMillis) {
-        randomAwait(cv, 0, maxMillis);
-    }
-    
-    //////////////////////////////////////////////////////////////////
-    
-    public static void sleepSeconds(double seconds, Object obj) {
-        int    millis;
-        
-        millis = (int)(seconds * 1000.0);
-        sleep(millis, obj);
+  protected static final Random random;
+
+  protected static final int DEF_EXCEPTION_PAUSE_MILLIS = 1000;
+
+  static {
+    random = new Random();
+  }
+
+  public static void sleepNanos(int nanos) {
+    try {
+      Thread.currentThread().sleep(0, nanos);
+    } catch (InterruptedException ie) {
     }
-    
-    public static void sleep(int millis, Object obj) {
-        if (millis > 0) {
-            synchronized (obj) {
-                try {
-                    obj.wait(millis);
-                } catch (InterruptedException ie) {
-                }
-            }
-        }
+  }
+
+  public static void sleepSeconds(double seconds) {
+    int millis;
+
+    millis = (int) (seconds * 1000.0);
+    sleep(millis);
+  }
+
+  public static void sleep(long millis) {
+    if (millis > Integer.MAX_VALUE) {
+      throw new RuntimeException("Invalid sleep");
+    } else {
+      sleep((int) millis);
     }
-    
-    public static void randomSleep(int minMillis, int maxMillis, Object obj) {
-        int    millis;
-        
-        if (maxMillis < minMillis) {
-            throw new RuntimeException("maxMillis < minMillis");
-        }
-        millis = minMillis + random.nextInt(Math.max(maxMillis - minMillis, 1));
-        sleep(millis, obj);
+  }
+
+  public static void sleep(int millis) {
+    if (millis > 0) {
+      try {
+        Thread.sleep(millis);
+      } catch (InterruptedException ie) {
+      }
+    }
+  }
+
+  public static void randomSleep(int minMillis, int maxMillis) {
+    int millis;
+
+    if (maxMillis < minMillis) {
+      throw new RuntimeException("maxMillis < minMillis");
     }
-    
-    public static void randomSleep(int maxMillis, Object obj) {
-        randomSleep(0, maxMillis, obj);
+    millis = minMillis + random.nextInt(Math.max(maxMillis - minMillis, 1));
+    sleep(millis);
+  }
+
+  public static void randomSleep(int maxMillis) {
+    randomSleep(0, maxMillis);
+  }
+
+  public static void await(Condition cv, long time, TimeUnit unit) {
+    if (time > 0) {
+      try {
+        cv.await(time, unit);
+      } catch (InterruptedException ie) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)